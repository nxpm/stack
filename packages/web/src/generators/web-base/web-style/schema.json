{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "WebStyle",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "appName": {
      "type": "string",
      "description": "Name of the app to connect to",
      "alias": "a"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "library": {
      "description": "The UI library",
      "type": "string",
      "default": "tailwind",
      "x-prompt": {
        "message": "Which type of library would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "bootstrap",
            "label": "Bootstrap"
          },
          {
            "value": "tailwind",
            "label": "Tailwind"
          }
        ]
      }
    }
  },
  "required": ["appName", "name"]
}
