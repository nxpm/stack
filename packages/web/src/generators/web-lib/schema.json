{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "id": "ApiLib",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "type": {
      "type": "string",
      "enum": ["data-access", "feature", "util"],
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": "What type would you like to create?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    }
  },
  "required": ["name"]
}
