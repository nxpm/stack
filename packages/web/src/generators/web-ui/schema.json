{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "id": "WebUi",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "library": {
      "description": "The UI library",
      "type": "string",
      "default": "tailwind",
      "x-prompt": {
        "message": "Which type of library would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "tailwind",
            "label": "Tailwind"
          }
        ]
      }
    }
  },
  "required": ["name"]
}
