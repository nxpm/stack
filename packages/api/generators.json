{
  "$schema": "http://json-schema.org/schema",
  "name": "api",
  "version": "0.0.1",
  "generators": {
    "api": {
      "factory": "./src/generators/api/generator-api#generatorApi",
      "schema": "./src/generators/api/schema.json",
      "description": "api generator"
    },
    "api-e2e": {
      "factory": "./src/generators/api-e2e/generator-api-e2e#generatorApiE2e",
      "schema": "./src/generators/api-e2e/schema.json",
      "description": "api-e2e generator"
    },
    "api-feature": {
      "factory": "./src/generators/api-feature/generator-api-feature#generatorApiFeature",
      "schema": "./src/generators/api-feature/schema.json",
      "description": "api-feature generator"
    },
    "api-lib": {
      "factory": "./src/generators/api-lib/generator-api-lib#generatorApiLib",
      "schema": "./src/generators/api-lib/schema.json",
      "description": "api-lib generator"
    }
  }
}
