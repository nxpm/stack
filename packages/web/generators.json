{
  "$schema": "http://json-schema.org/schema",
  "name": "web",
  "version": "0.0.1",
  "generators": {
    "web": {
      "factory": "./src/generators/web/generator-web#generatorWeb",
      "schema": "./src/generators/web/schema.json",
      "description": "web generator"
    },
    "web-base": {
      "factory": "./src/generators/web-base/generator-web-base#generatorWebBase",
      "schema": "./src/generators/web-base/schema.json",
      "description": "web-base generator"
    },
    "web-feature": {
      "factory": "./src/generators/web-feature/generator-web-feature#generatorWebFeature",
      "schema": "./src/generators/web-feature/schema.json",
      "description": "web-feature generator"
    },
    "web-lib": {
      "factory": "./src/generators/web-lib/generator-web-lib#generatorWebLib",
      "schema": "./src/generators/web-lib/schema.json",
      "description": "web-lib generator"
    },
    "web-ui": {
      "factory": "./src/generators/web-ui/generator-web-ui#generatorWebUi",
      "schema": "./src/generators/web-ui/schema.json",
      "description": "web-ui generator"
    },
    "web-ui-lib": {
      "factory": "./src/generators/web-ui-lib/generator-web-ui-lib#generatorWebUiLib",
      "schema": "./src/generators/web-ui-lib/schema.json",
      "description": "web-ui-lib generator"
    }
  }
}
