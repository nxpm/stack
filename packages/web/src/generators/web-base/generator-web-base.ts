import { Tree } from '@nrwl/devkit'
import { WebBaseGeneratorSchema } from './schema'
import { generateWebAssets } from './generate-web-assets'
import { generateWebStyle } from './generate-web-style'
import { generateWebEnvironments } from './generate-web-environments'

export async function generatorWebBase(host: Tree, options: WebBaseGeneratorSchema) {
  await generateWebAssets(host, {
    ...options,
    directory: options.directory || options.webName,
    name: 'assets',
  })
  await generateWebEnvironments(host, {
    ...options,
    directory: options.directory || options.webName,
    name: 'environments',
  })
  await generateWebStyle(host, {
    ...options,
    directory: options.directory || options.webName,
    name: 'style',
  })
}
