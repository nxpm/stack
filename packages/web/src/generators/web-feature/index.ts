export * from './generate-web-feature-about'
export * from './generate-web-feature-account'
export * from './generate-web-feature-admin'
export * from './generate-web-feature-auth'
export * from './generate-web-feature-core'
export * from './generate-web-feature-dashboard'
export * from './generate-web-feature-layout'
export * from './generate-web-feature-shell'
