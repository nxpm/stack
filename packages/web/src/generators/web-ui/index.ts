export * from './generate-web-ui-lib-button'
export * from './generate-web-ui-lib-form'
export * from './generate-web-ui-lib-icon'
export * from './generate-web-ui-lib-loader'
export * from './generate-web-ui-lib-page'
export * from './generate-web-ui-lib-page-header'
export * from './generate-web-ui-lib-sidebar-page'
export * from './generate-web-ui-lib-table'
export * from './generate-web-ui-lib-toast'
