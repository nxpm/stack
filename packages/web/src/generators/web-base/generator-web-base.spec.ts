import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing'
import { readJson, readProjectConfiguration, Tree } from '@nrwl/devkit'

import { generatorWebBase } from './generator-web-base'
import { WebBaseGeneratorSchema } from './schema'
import { applicationGenerator } from '@nrwl/angular/generators'

describe('web-base generator', () => {
  let appTree: Tree
  const options: WebBaseGeneratorSchema = { name: 'assets', webName: 'web-test' }

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace()
  })

  it('should run successfully', async () => {
    await applicationGenerator(appTree, { name: options.webName })
    await generatorWebBase(appTree, options)

    const nxJson = readJson(appTree, 'nx.json')
    expect(nxJson).toMatchSnapshot()

    const configApp = readProjectConfiguration(appTree, `${options.webName}`)
    expect(configApp).toMatchSnapshot()

    const configAssets = readProjectConfiguration(appTree, `${options.webName}-assets`)
    expect(configAssets).toMatchSnapshot()

    const configEnvironments = readProjectConfiguration(appTree, `${options.webName}-environments`)
    expect(configEnvironments).toMatchSnapshot()
  })
})
