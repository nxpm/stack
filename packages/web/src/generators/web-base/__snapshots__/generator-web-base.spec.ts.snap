// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web-base generator should run successfully 1`] = `
Object {
  "affected": Object {
    "defaultBase": "master",
  },
  "npmScope": "proj",
  "projects": Object {
    "web-test": Object {
      "tags": Array [],
    },
    "web-test-assets": Object {
      "tags": Array [],
    },
    "web-test-e2e": Object {
      "implicitDependencies": Array [
        "web-test",
      ],
      "tags": Array [],
    },
    "web-test-environments": Object {
      "tags": Array [],
    },
    "web-test-style": Object {
      "tags": Array [],
    },
  },
  "tasksRunnerOptions": Object {
    "default": Object {
      "options": Object {
        "cacheableOperations": Array [
          "build",
          "lint",
          "test",
          "e2e",
        ],
      },
      "runner": "@nrwl/workspace/tasks-runners/default",
    },
  },
}
`;

exports[`web-base generator should run successfully 2`] = `
Object {
  "prefix": "proj",
  "projectType": "application",
  "root": "apps/web-test",
  "sourceRoot": "apps/web-test/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "configurations": Object {
        "development": Object {
          "buildOptimizer": false,
          "extractLicenses": false,
          "namedChunks": true,
          "optimization": false,
          "sourceMap": true,
          "vendorChunk": true,
        },
        "production": Object {
          "budgets": Array [
            Object {
              "maximumError": "1mb",
              "maximumWarning": "500kb",
              "type": "initial",
            },
            Object {
              "maximumError": "4kb",
              "maximumWarning": "2kb",
              "type": "anyComponentStyle",
            },
          ],
          "fileReplacements": Array [
            Object {
              "replace": "libs/web-test/environments/src/environments/environment.ts",
              "with": "libs/web-test/environments/src/environments/environment.prod.ts",
            },
          ],
          "outputHashing": "all",
        },
      },
      "defaultConfiguration": "production",
      "executor": "@angular-devkit/build-angular:browser",
      "options": Object {
        "assets": Array [
          Object {
            "glob": "favicon.ico",
            "input": "libs/web-test/assets/src",
            "output": "./",
          },
          Object {
            "glob": "**/*",
            "input": "libs/web-test/assets/src/assets",
            "output": "assets",
          },
        ],
        "index": "apps/web-test/src/index.html",
        "main": "apps/web-test/src/main.ts",
        "outputPath": "dist/apps/web-test",
        "polyfills": "apps/web-test/src/polyfills.ts",
        "scripts": Array [],
        "styles": Array [
          "apps/web-test/src/styles.css",
        ],
        "tsConfig": "apps/web-test/tsconfig.app.json",
      },
      "outputs": Array [
        "{options.outputPath}",
      ],
    },
    "extract-i18n": Object {
      "executor": "@angular-devkit/build-angular:extract-i18n",
      "options": Object {
        "browserTarget": "web-test:build",
      },
    },
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "apps/web-test/src/**/*.ts",
          "apps/web-test/src/**/*.html",
        ],
      },
    },
    "serve": Object {
      "configurations": Object {
        "development": Object {
          "browserTarget": "web-test:build:development",
        },
        "production": Object {
          "browserTarget": "web-test:build:production",
        },
      },
      "defaultConfiguration": "development",
      "executor": "@angular-devkit/build-angular:dev-server",
    },
    "test": Object {
      "executor": "@nrwl/jest:jest",
      "options": Object {
        "jestConfig": "apps/web-test/jest.config.js",
        "passWithNoTests": true,
      },
      "outputs": Array [
        "coverage/apps/web-test",
      ],
    },
  },
}
`;

exports[`web-base generator should run successfully 3`] = `
Object {
  "prefix": "proj",
  "projectType": "library",
  "root": "libs/web-test/assets",
  "sourceRoot": "libs/web-test/assets/src",
  "tags": Array [],
  "targets": Object {},
}
`;

exports[`web-base generator should run successfully 4`] = `
Object {
  "prefix": "proj",
  "projectType": "library",
  "root": "libs/web-test/environments",
  "sourceRoot": "libs/web-test/environments/src",
  "tags": Array [],
  "targets": Object {
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "libs/web-test/environments/src/**/*.ts",
          "libs/web-test/environments/src/**/*.html",
        ],
      },
    },
    "test": Object {
      "executor": "@nrwl/jest:jest",
      "options": Object {
        "jestConfig": "libs/web-test/environments/jest.config.js",
        "passWithNoTests": true,
      },
      "outputs": Array [
        "coverage/libs/web-test/environments",
      ],
    },
  },
}
`;
